src/nvim/tui/tui.c

```diff
@@ -641,12 +646,21 @@ static void update_size(UI *ui)
 {
   TUIData *data = ui->data;
   int width = 0, height = 0;
-  // 1 - try from a system call(ioctl/TIOCGWINSZ on unix)
+
+  // 1 - if starting and the 'columns' and 'lines' options are not
+  // the defaults, use their values
+  if (starting != 0 && (Columns != DFLT_COLS || Rows != DFLT_LINES)) {
+    width = (int)Columns;
```

Comment:
`Columns` is of type `long`, so isn't guaranteed to fit into an `int`. I recall it being discussed earlier that `Columns` and `Rows` should really be of type `int`, but that might be rather invasive of a refactoring.